Source/FieldSolver/FiniteDifferenceSolver/HybridPICSolveE.cpp

```diff
@@ -95,11 +95,13 @@ void FiniteDifferenceSolver::CalculateCurrentAmpereCylindrical (
         Array4<Real> const& Bt = Bfield[1]->array(mfi);
         Array4<Real> const& Bz = Bfield[2]->array(mfi);
 
-#ifdef AMREX_USE_EB
-        amrex::Array4<amrex::Real> const& lr = edge_lengths[0]->array(mfi);
-        amrex::Array4<amrex::Real> const& lt = edge_lengths[1]->array(mfi);
-        amrex::Array4<amrex::Real> const& lz = edge_lengths[2]->array(mfi);
-#endif
+        amrex::Array4<amrex::Real> lr, lt, lz;
+
+        if (eb_enabled) {
+            lr = edge_lengths[0]->array(mfi);
+            lt = edge_lengths[1]->array(mfi);
+            lz = edge_lengths[2]->array(mfi);
+        }
```

Comment:
It would be nice if we can always have a well defined EBFactory which just doesn't cover any of the domain when it is not wanted. Then the `edge_lengths` and similar multifabs could be properly defined in all cases and we wouldn't have to have the `if (eb_enabled)` block. Similarly lime 131 below could simply be:
```
if (lr(i, j, 0) <= 0) return;
```
and the same for all the other similar occurrences.

Reply:
That would waste memory.

Reply:
Unless we would implement an `Array4` variant that returns 0 outside of the covered domain, which we could use elsewhere as well. But that then probably leads to register bloat (unless we implement as in `mdspan` variants that have known compile-time sizes).